as/itests/org.jboss.tools.as.ui.bot.itests/src/org/jboss/tools/as/ui/bot/itests/parametized/server/ServerAdaptersTest.java

```diff
@@ -97,21 +95,14 @@ public void setupLocalServerAdapter() {
 			setupRuntime(serverW);
 
 			serverW.finish();
-		} catch (RuntimeException e) {
+		} catch (AssertionError | RuntimeException e) {
 			try {
 				serverW.cancel();
 			} catch (RedDeerException ex) {
-				throw e;
+				throw ex;
```

Comment:
I would log the exception and print its stack trace and then let it continue to parent block to throw Assertion or runtime exception. This way you would not lost the former exception when trying to cancel the dialog.